src/cdk/a11y/aria-describer.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Injectable, Optional, Renderer2} from '@angular/core';
+import {Platform} from '@angular/cdk/platform';
+import {addAriaReferencedId, removeAriaReferencedId} from './aria-reference';
+
+/**
+ * Interface used to register message elements and keep a count of how many registrations have
+ * the same message and the reference to the message element used for the aria-describedby.
+ */
+export interface RegisteredMessage {
+  messageElement: HTMLElement;
+  hostElements: HTMLElement[];
```

Comment:
Why do you need to keep track of these elements? I'm thinking this may cause a memory leak if references to elements are preserved here when the elements have been removed.

Could you accomplish what you need by adding an attribute to the described element?